### ISSUE TITLE
HTTP2 support

### ISSUE THREAD

**Issue by lpil** [elli-lib/elli#78]
It's be great to have HTTP2 support in Elli!

This seems like quite an ambitious addition but I thought I'd open an issue anyway :)

**Comment by tsloughter** [comment]
@lpil I don't think it should ever be added to elli itself, which is why I created https://github.com/elli-lib/elli_chatterbox.

Obviously issue being they don't have a shared form of request and configuration, all they share is the tcp acceptor which based on if you send a h2 upgrade in ssl it goes to http2.

But this could be built on to provide a shared interface for both while keeping it not added it directly into elli itself.

**Comment by yurrriq** [comment]
:+1: to shared interface and an example in https://github.com/elli-lib/elli-examples

**Comment by yurrriq** [comment]
Opened https://github.com/elli-lib/elli-examples/issues/8

**Comment by yurrriq** [comment]
Is there some way we could use the Plug design for the shared interface?

**Comment by tsloughter** [comment]
I wouldn't.

But I'd be open to reviving https://github.com/tsloughter/elna, just not sure if there is a user base out there. 

Also, I've been meaning to merge in an acceptor_pool interface to chatterbox and submit one for elli, then they'll both be easily combined, tho with their own handlers, so only useful for those who have a specific use-case where they want the http1/http2 to be handled by separate modules.
